react/features/remote-video-menu/components/native/PinButton.js

```diff
@@ -0,0 +1,80 @@
+// @flow
+
+import { translate } from '../../../base/i18n';
+import { pinParticipant } from '../../../base/participants';
+import { connect } from '../../../base/redux';
+import { AbstractButton, type AbstractButtonProps } from '../../../base/toolbox';
+import { setTileView } from '../../../video-layout';
+
+export type Props = AbstractButtonProps & {
+
+    /**
+     * True if tile view is currently enabled.
+     */
+    _tileViewEnabled: boolean,
+
+    /**
+     * The redux {@code dispatch} function.
+     */
+    dispatch: Function,
+
+    /**
+     * The ID of the participant that this button is supposed to pin.
+     */
+    participantID: string,
+
+    /**
+     * The function to be used to translate i18n labels.
+     */
+    t: Function
+};
+
+/**
+ * A remote video menu button which pins a participant and exist the tile view.
+ */
+class PinButton extends AbstractButton<Props, *> {
+    accessibilityLabel = 'toolbar.accessibilityLabel.show';
+    iconName = 'icon-enlarge';
+    label = 'videothumbnail.show';
+
+    /**
+     * Overrides {@code AbstractButton#_shouldRender}.
+     *
+     * @inheritdoc
+     */
+    _shouldRender() {
+        return this.props._tileViewEnabled;
+    }
+
+    /**
+     * Handles clicking / pressing the button, and kicks the participant.
+     *
+     * @private
+     * @returns {void}
+     */
+    _handleClick() {
+        const { dispatch } = this.props;
+
+        // Pin participant
+        dispatch(pinParticipant(this.props.participantID));
+
+        // Then exit tile view to show
+        dispatch(setTileView(false));
```

Comment:
I'm surprised pinning a participant doesn't automatically exit tile view.

Reply:
I guess that would involve other questions, like should we be able to get into tile view when there is a pinned participant? If we get into tile view, should we unpin all participants?

Reply:
Should we be able to get into tile view when there is a pinned participant? Yes.
If we get into tile view, should we unpin all participants? Yes.

What surprises me is the participants middleware isn't automatically firing to undo tile view. I see there is middleware for it.

Reply:
hmm, maybe it's worth exploring that situation... give me a minute

Reply:
Yep, that works perfectly, thanks for the suggestion!